staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp.go

```diff
@@ -151,15 +161,33 @@ func tokenSource(isCmd bool, gcpConfig map[string]string) (oauth2.TokenSource, e
 		return newCmdTokenSource(cmd, args, gcpConfig["token-key"], gcpConfig["expiry-key"], gcpConfig["time-fmt"]), nil
 	}
 
-	// Google Application Credentials-based token source
 	scopes := parseScopes(gcpConfig)
+	// Service Account Credentials in JSON-based token source
+	if isServiceAcctCredTokenSource(gcpConfig) {
+		ts, err := tokenSourceFromJSON(context.Background(), gcpConfig["service-acct"], scopes...)
```

Comment:
nit: the key string "service-acct" is used twice in the code, better to use a const for it.

Reply:
I pattern matched from the key string `"cmd-path"` which is also used twice and is not a const.

Would you like me to pull both out into constants?